### PR TITLE
3414894965 base subscription events

### PR DESCRIPTION
**Pull request by agpmercado**
Hey Oscar, 

Raising my Pull Request although I know there may be a number of changes for Access layer objects that I want to discuss further with you during Office Hours tomorrow.

### CODE REVIEW

File: models/access/dim/dim_product.sql [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    p.product_hero_id,
+    ph.product_hero_name,
+    ph.product_hero_category_id,
+    phc.product_hero_category_name,
+    p.price_amount, 
+    p.active_flag, 
+    p.active_customer_flag, 
+    ph.is_collection_flag, 
+    count(ao.add_on_id) add_on_quantity,
+    sum(case when ao.add_on_id = 1 then 1 else 0 end) as welcome_pack,
```
Comment (lukersmithoscar) [inline models/access/dim/dim_product.sql:17]: hey @agpmercado, let's talk about these case statements.

File: models/access/fact/base/base_user_subscription_event.sql [hunk 2]
```diff
@@ -0,0 +1,218 @@
+
```
Comment (lukersmithoscar) [inline models/access/fact/base/base_user_subscription_event.sql:1]: We can refactor this as well

File: models/staging/portal/stg_portal_add_on.sql [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        name as add_on_name,
+        display_name as add_on_display_name,
+        customer_display_name,
+        description as add_on_description,
+        boxing_interface_display_name,
+        boxing_interface_section as boxing_interface_section_text,
+        active as active_flag,
+        boxing_interface_scannable as boxing_interface_scannable_flag,
+        image_reference,
+        barcodes as barcode,
```
Comment (lukersmithoscar) [inline models/staging/portal/stg_portal_add_on.sql:21]: doesn't conform to our naming standard

File: models/staging/portal/stg_portal_order_inventory.sql [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        total_lamb_override,
+        total_sensitive_chicken,
+        total_sensitive_chicken_override,
+        total_turkey,
+        total_turkey_override,
+        case when (coalesce(total_kangaroo_override,0) + coalesce(total_turkey_override,0) + coalesce(total_lamb_override,0) + coalesce(total_beef_override,0) +
+                    coalesce(total_chicken_override,0) + coalesce(total_sensitive_chicken_override,0) + coalesce(total_fish_override,0)) > 0 
+             then true 
+             else false 
+        end as is_override_flag
```
Comment (lukersmithoscar) [inline models/staging/portal/stg_portal_order_inventory.sql:39]: naming standard

File: models/staging/portal/stg_portal_order_inventory.sql [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        total_fish,
+        total_fish_override,
+        total_kangaroo,
+        total_kangaroo_override,
+        total_lamb,
+        total_lamb_override,
+        total_sensitive_chicken,
+        total_sensitive_chicken_override,
+        total_turkey,
+        total_turkey_override,
```
Comment (lukersmithoscar) [inline models/staging/portal/stg_portal_order_inventory.sql:34]: naming standard

File: models/staging/portal/stg_portal_order_inventory.sql [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+        daily_grams,
+        daily_grams_override,
+        meal_plan_id,
+        order_id,
+        pouch_size_id,
+        pouches_total,
+        pouches_total_override,
+        production_cycle_id,
+        total_beef,
+        total_beef_override,
```
Comment (lukersmithoscar) [inline models/staging/portal/stg_portal_order_inventory.sql:22]: naming standard

File: models/staging/portal/stg_portal_order_inventory.sql [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+),
+
+checkoverrides as ( 
+
+    select
+        id,
+        _fivetran_synced,
+        animal_id,
+        daily_grams,
+        daily_grams_override,
```
Comment (lukersmithoscar) [inline models/staging/portal/stg_portal_order_inventory.sql:14]: naming standard

File: models/staging/portal/stg_portal_product.sql [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+        id as product_id,
+        product_hero_id,
+        netsuite_id,
+        name as product_name,
+        display_name as product_display_name,
+        customer_display_name,
+        customer_sub_display_name,
+        active as active_flag,
+        active_customer as active_customer_flag,
+        sku_size,
```
Comment (lukersmithoscar) [inline models/staging/portal/stg_portal_product.sql:19]: what is this? naming standard

**Review by lukersmithoscar — CHANGES_REQUESTED** [review]
Hey @agpmercado, great start, its a tonne of work! 
Lets book a session to run over the feedback, it will be easier in person.
I am super excited because now is the time I get to show you the **exciting** part of dbt and how it will help you.

**Comment by agpmercado** [comment]
> 

Hey @lukersmithoscar , thanks for reviewing them! Although please note this was raised before our Office Session yesterday so I know a bunch of stuff will change in the Access layers part. I'll revise the comments on the staging layers then set-up a session as you have suggested. Excited to learn more about it as well! :)

File: models/staging/portal/stg_portal_order_inventory.sql (hunk 7 above)
Comment (agpmercado) [reply models/staging/portal/stg_portal_order_inventory.sql:14]: I have renamed this in final select statement! should we also alias it in CTEs?

File: models/access/dim/dim_product.sql (hunk 1 above)
Comment (agpmercado) [reply models/access/dim/dim_product.sql]: I have deleted these for now as this is for another data model & since we're not doing dimensions atm

**Review by lukersmithoscar — APPROVED** [review]

**Pull request merged**